actors/common/src/main/java/org/sunbird/learner/actors/data/security/manager/DataSecurityBackgroundActor.java

```diff
@@ -1,65 +1,93 @@
 
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import org.apache.commons.collections.CollectionUtils;
 import org.sunbird.actor.core.BaseActor;
 import org.sunbird.actor.router.ActorConfig;
 import org.sunbird.cassandra.CassandraOperation;
 import org.sunbird.common.models.response.Response;
 import org.sunbird.common.models.util.ActorOperations;
 import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.LoggerEnum;
 import org.sunbird.common.models.util.ProjectLogger;
 import org.sunbird.common.models.util.ProjectUtil;
 import org.sunbird.common.request.Request;
 import org.sunbird.helper.ServiceFactory;
 import org.sunbird.learner.util.UserUtility;
 import org.sunbird.learner.util.Util;
 
+/** Background sync of data between Cassandra and Elastic Search. */
 @ActorConfig(
-  tasks = {"encryptUserData", "decryptUserData"},
-  asyncTasks = {}
+  tasks = {},
+  asyncTasks = {"backgroundEncryption", "backgroundDecryption"}
 )
-public class UserDataEncryptionDecryptionServiceActor extends BaseActor {
+public class DataSecurityBackgroundActor extends BaseActor {
 
   private CassandraOperation cassandraOperation = ServiceFactory.getInstance();
   private Util.DbInfo usrDbInfo = Util.dbInfoMap.get(JsonKey.USER_DB);
   private Util.DbInfo addrDbInfo = Util.dbInfoMap.get(JsonKey.ADDRESS_DB);
 
   @Override
-  public void onReceive(Request actorMessage) throws Throwable {
-    if (actorMessage
-        .getOperation()
-        .equalsIgnoreCase(ActorOperations.ENCRYPT_USER_DATA.getValue())) {
-      encryptUserData(actorMessage);
-    } else if (actorMessage
-        .getOperation()
-        .equalsIgnoreCase(ActorOperations.DECRYPT_USER_DATA.getValue())) {
-      decryptUserData(actorMessage);
-    } else {
-      onReceiveUnsupportedOperation(actorMessage.getOperation());
+  public void onReceive(Request request) throws Throwable {
+    String operation = request.getOperation();
+    switch (operation) {
+      case "backgroundEncryption":
+        backgroundEncrypt(request);
+        break;
+      case "backgroundDecryption":
+        backgroundDecrypt(request);
+        break;
+      default:
+        onReceiveUnsupportedOperation("EsSyncBackgroundActor");
+        break;
     }
   }
 
-  private void decryptUserData(Request actorMessage) {
-    ProjectLogger.log(
-        "DecryptUserData API called by " + actorMessage.getRequest().get(JsonKey.REQUESTED_BY));
-    long start = System.currentTimeMillis();
-    Response resp = new Response();
-    resp.put(JsonKey.RESPONSE, JsonKey.SUCCESS);
-    sender().tell(resp, self());
-    Response response =
-        cassandraOperation.getAllRecords(usrDbInfo.getKeySpace(), usrDbInfo.getTableName());
-    List<Map<String, Object>> userList = (List<Map<String, Object>>) response.get(JsonKey.RESPONSE);
+  private void backgroundEncrypt(Request request) {
+    List<Map<String, Object>> userDetails = getUserDetails(request);
```

Comment:
following 2 functions are template - except the condition and call to function -> can u make it generic via lambda or something?

or create private function which based on boolean - either encrypts or decrypts the data -> base funciton shud be common for both cases